### PR TITLE
feat: added eslint, stylelint, and prettier

### PR DESCRIPTION
**Pull request by jimbateson**
- ESlint config and package scripts
- Stylelint config and package scripts
- Prettier config and ignore with a format package script
- RUNBOOK documentation

Is the runbook where we want to put these instructions?

Fixes issue #8

### CODE REVIEW

**Comment by ahallicks** [comment]
Yeah, docs should go in the runbook if they're for setting up the project 😄

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        "plugin:@typescript-eslint/recommended",
+        "plugin:react/recommended",
+        "plugin:react-hooks/recommended",
+        "plugin:jsx-a11y/recommended",
+      ],
+      rules: {
+        "react/no-unescaped-entities": 0,
+        "react/prop-types": "off", // ts types used instead
+        "react/react-in-jsx-scope": "off", // no need to import React when using next
+        "jsx-a11y/anchor-is-valid": "off", // not compatible with next Link
```
Comment (ahallicks) [inline .eslintrc.js:40]: I think this might not be needed any more as Next Link doesn't require a nested `<a>` now!

**Comment by jimbateson** [comment]
Linting rules are pretty relaxed for now I think. These are probably things that can be changed on a per-project basis and developers preference, but just something consistent for us to start with.

**Review by ahallicks — APPROVED** [review]

**Review by lucilevacquie — DISMISSED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (jimbateson) [reply .eslintrc.js:40]: Ah ok, I thought it might be checking all anchors in JSX in general not just the `<Link />` ones?

Comment (jimbateson) [reply .eslintrc.js:40]: Ohhh, you mean it should be on now sorry!?

Comment (ahallicks) [reply .eslintrc.js:40]: Yeah. I think it was turned off because of the weird way that Next used to do anchors:
```
<Link><a>Text</a></Link>
```

Comment (jimbateson) [reply .eslintrc.js]: Should be sorted now, I've removed as it's included in `eslint:recommended`.

**Review by ahallicks — APPROVED** [review]

**Review by ahallicks — APPROVED** [review]

**Pull request merged**